--- elfutils-0.155/src/nm.c    2013-03-11 22:36:11.000000000 -0500
+++ elfutils-0.155/src/nm.c    2013-03-11 22:46:09.000000000 -0500
@@ -777,8 +777,9 @@
                    gelf_getshdr (scn, &shdr_mem)->sh_name);
       if (unlikely (name == NULL))
     {
-      name = alloca (sizeof "[invalid sh_name 0x12345678]");
-      snprintf (name, sizeof name, "[invalid sh_name %#" PRIx32 "]",
+      const size_t bufsz = sizeof "[invalid sh_name 0x12345678]";
+      name = alloca (bufsz);
+      snprintf (name, bufsz, "[invalid sh_name %#" PRIx32 "]",
             gelf_getshdr (scn, &shdr_mem)->sh_name);
     }
       scnnames[elf_ndxscn (scn)] = name;
